### PR TITLE
Add loop problem practice resource

### PR DESCRIPTION
**Pull request by kariabancroft**
The purpose of this resource is in preparation for our conversation in Week 4 about Enumerables. For each of these problems, the students can solve them using the "regular" `each` loops that they know about. We'd assign this as HW prior to lecturing on Enumerables.

The idea is that once they do that, then we can introduce the Enumerable for each one of these problems to show how useful they can be and how they can simplify the code.

I think the [Enumerable lecture](https://github.com/Ada-Developers-Academy/textbook-curriculum/blob/master/02-intermediate-ruby/07-enumerable.md) right now is a bit too arbitrary about learning what each method does, rather than learning _why_ we'd want to use each method. The idea is that using these problems, we'll be able to more clearly explain the _why_.

### CODE REVIEW

**Comment by CheezItMan** [comment]
Would a link to reference material on the enumerable be appropriate?

File: loop-problem-practice.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Loop Problem Practice
+
+For each of the following problems, write the code to solve the problem.
+
+1. You are given an array of `String` objects, stored in a variable called `my_strings`. You want to use this array to create a new array of `Integer` objects, which represents the `length` of each string. We want to store the resulting array in a variable called `string_lengths`.
+
+<!-- MAP/COLLECT -->
```
Comment (CheezItMan) [inline loop-problem-practice.md]: Because of this, maybe the bullets are all numbered '1'

**Comment by kariabancroft** [comment]
@CheezItMan No, this is for _before_ we learn enumerables. Folks will solve these problems and then we'll show them the solution using an Enumerable together in class (or again as HW after the lecture).

File: loop-problem-practice.md (hunk 1 above)
Comment (kariabancroft) [reply loop-problem-practice.md]: Good call! Fixed

**Comment by droberts-ada** [comment]
```

                   _________|\
              _.--| Ship It |:
             <____|.--------||
                        .---''---,
                         ;..__..'    _...
                       ,'/  ;|/..--''    \
                     ,'_/.-/':            :
                _..-'''/  /  |  \    \   _|/|
               \      /-./_ \;   \    \,;'   \
               ,\    / \:  `:\    \   //    `:`.
             ,'  \  /-._;   | :    : ::    ,.   .
           ,'     ::   /`-._| |    | || ' :  `.`.)
        _,'       |;._:: |  | |    | `|   :    `'
      ,'   `.     /   |`-:_ ; |    |  |  : \
      `--.   )   /|-._:    :          |   \ \
         /  /   :_|   ;`-._;   __..--';    : :
        /  (    ;|;-./_  _/.-:'o |   /     ' |
       /  , \._/_/_./--''/_|:|___|_,'        |
      :  /   `'-'--'----'---------'          |
      | :     O ._O   O_. O ._O   O_.      ; ;
      : `.      //    //    //    //     ,' /
    ~~~`.______//____//____//____//_______,'~
              //    //~   //    //
       ~~   _//   _//   _// ~ _//     ~
     ~     / /   / /   / /   / /  ~      ~~
          ~~~   ~~~   ~~~   ~~~
```

**Review by droberts-sea — COMMENTED** [review]
```

                   _________|\
              _.--| Ship It |:
             <____|.--------||
                        .---''---,
                         ;..__..'    _...
                       ,'/  ;|/..--''    \
                     ,'_/.-/':            :
                _..-'''/  /  |  \    \   _|/|
               \      /-./_ \;   \    \,;'   \
               ,\    / \:  `:\    \   //    `:`.
             ,'  \  /-._;   | :    : ::    ,.   .
           ,'     ::   /`-._| |    | || ' :  `.`.)
        _,'       |;._:: |  | |    | `|   :    `'
      ,'   `.     /   |`-:_ ; |    |  |  : \
      `--.   )   /|-._:    :          |   \ \
         /  /   :_|   ;`-._;   __..--';    : :
        /  (    ;|;-./_  _/.-:'o |   /     ' |
       /  , \._/_/_./--''/_|:|___|_,'        |
      :  /   `'-'--'----'---------'          |
      | :     O ._O   O_. O ._O   O_.      ; ;
      : `.      //    //    //    //     ,' /
    ~~~`.______//____//____//____//_______,'~
              //    //~   //    //
       ~~   _//   _//   _// ~ _//     ~
     ~     / /   / /   / /   / /  ~      ~~
          ~~~   ~~~   ~~~   ~~~
```

**Review by droberts-sea — APPROVED** [review]

**Pull request merged**